### PR TITLE
Update documentation and use clang-format

### PR DESCRIPTION
**Pull request by abeaumont**

### CODE REVIEW

File: src/uast.h [hunk 1]
```diff
@@ -1,54 +1,47 @@
...
 #include "node_iface.h"
+#include "nodes.h"
 
-// NodeApi stores all general context required for calling the libuast's api
-// It must be initialized with `NewNodeApi(NodeIface)` passing a valid
-// implementation of the interface
-// Once it is not used anymore, it shall be released by using `FreeNodeApi()`
+// Uast stores the general context required for library functions.
+// It must be initialized with `UastNew` passing a valid implementation of the
+// NodeIface interface.
```
Comment (juanjux) [inline src/uast.h]: \`NodeIface\` (with the fixed size format backticks).

File: src/uast.h [hunk 2]
```diff
@@ -1,54 +1,47 @@
...
+// Once it is not used anymore, it shall be released calling `UastFree`.
 typedef struct Uast Uast;
 
-// NodeAPI needs a node implementation in other to work. This is needed
+// Uast needs a node implementation in other to work. This is needed
```
Comment (juanjux) [inline src/uast.h]: in *order* to work

File: src/uast.h [hunk 3]
```diff
@@ -1,54 +1,47 @@
...
-// following way:
+// A node will be mapped to the following XML representation:
 // ```
 // <{{INTERNAL_TYPE}} token={{TOKEN}} role{{ROLE[n]}} prop{{PROP[n]}}>
 //   ... children
 // </{{INTERNAL_TYPE}}>
 // ```
 //
-// Real python example:
+// An exmpale in Python:
```
Comment (juanjux) [inline src/uast.h]: example

**Comment by abeaumont** [comment]
Fixed, thanks!

**Pull request merged**